### PR TITLE
dtoh: Make implementation private and add some DDOC

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Also moved several utility methods s.t. the public interface is at the beginning of the file

### CODE REVIEW

File: src/dmd/dtoh.d [hunk 1]
```diff
@@ -251,27 +199,58 @@ public:
...
     AST.AggregateDeclaration adparent;
+
+    /// Enclosing template declaration
     AST.TemplateDeclaration tdparent;
+
+    /// Identifier of the currently visited `VarDeclaration`
+    /// (required to write variables of funtion pointers)
     Identifier ident;
-    LINK linkage = LINK.d;
-    bool forwardedAA;
```
Comment (MoonlightSentinel) [inline src/dmd/dtoh.d:266]: Removed this attribute because it is unused.

File: src/dmd/frontend.h [hunk 2]
```diff
@@ -3424,89 +3424,6 @@ class TemplateMixin final : public TemplateInstance
 
 extern void genCppHdrFiles(Array<Module* >& ms);
 
-class ToCppBuffer final : public Visitor
```
Comment (MoonlightSentinel) [inline src/dmd/frontend.h:3427]: This declaration is now longer emitted because it is `private`

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/frontend.h (hunk 2 above)
Comment (RazvanN7) [reply src/dmd/frontend.h:3427]: I noticed that some of the class's methods are made `private`. Is that really necessary, since the class is `private`?

Comment (MoonlightSentinel) [reply src/dmd/frontend.h:3427]: No. I just marked the few remaining methods as `private` s.t. it's consistently used for all internal methods.

**Pull request merged**